static/extensions/orengdog/Music to numbers.js

```diff
@@ -0,0 +1,323 @@
+(async function(Scratch) {
+    const variables = {
+        volume: 1.0,
+        pitchShift: 0,
+        leftRightValue: 0,
+        selectedEffect: "none",
+        isPlaying: false,
+        instrument: "sine",
+        instrumentID: 0,
+        instrumentName: "without instrument",
+        pauseDuration: 0.5
+    };
+
+    class MusicGenerator {
+        constructor() {
+            this.audioContext = new (window.AudioContext || window.webkitAudioContext)();
```

Comment:
make sure to register this audioContext to PM.
you'll need to connect stuff that originally connects to `audioContext.destination` to a main gain node that **is** connected to `audioContext.destination`, you already make gain nodes for playing notes so you can just make a main gain node here

you can then register the context with this function:
```js
runtime.registerExtensionAudioContext("EXTENSION_ID", audioContext, gainNode);
```
make sure to replace EXTENSION_ID with your extension's ID.

this will make the pause button and volume slider work with the extension, and future addons or extensions will be able to modify it if needed